compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -1424,7 +1467,9 @@ fn panic_on_forbidden_read<D: Deps>(data: &DepGraphData<D>, dep_node_index: DepN
     // First try to find the dep node among those that already existed in the
     // previous session and has been marked green
     for prev_index in data.colors.values.indices() {
-        if data.colors.current(prev_index) == Some(dep_node_index) {
+        if let DepNodeColor::Green(c) = data.colors.current(prev_index)
```

Comment:
```suggestion
        if data.colors.current(prev_index) == DepNodeColor::Green(dep_node_index) {
```
Or remove several derived impls on `DepNodeColor` as unnecessary.